LayerClient/LayerClient.py

```diff
@@ -0,0 +1,519 @@
+    case, the encoding field must be set.
+    """
+
+    def __init__(self, body, mime=MIME_TEXT_PLAIN, encoding=None):
+        self.body = body
+        self.mime_type = mime
+        self.encoding = encoding
+
+    @staticmethod
+    def from_dict(dict_data):
+        return MessagePart(
+            dict_data.get('body'),
+            dict_data.get('mime_type'),
+            dict_data.get('encoding'),
+        )
+
+    def __repr__(self):
+        return (
+            '<LayerClient.MessagePart "{body}"{mime}{encoding}>'
+            .format(
+                body=self.body,
+                mime=' Content-Type: {0}'.format(self.mime_type),
+                encoding=(
+                    ' Encoding: {0}'.format(self.encoding) if self.encoding
+                    else ''
+                )
+            )
+        )
+
+    def as_dict(self):
+        data = {
+            'body': self.body,
+            'mime_type': self.mime_type,
+        }
+        if self.encoding:
+            data['encoding'] = self.encoding
+
+        return data
+
+
+class PushNotification:
+    """
+    Details for a push notification sent as part of a conversation message.
+    Each push notification must have a body.
+    Sound and recipients are optional. For Android, the sound parameter is
+    simply sent to the client as a string.
+    The recipients field is a map of user id to PushNotification object,
+    allowing for one push notification to have custom settings for certain
+    users. PushNotification instances used as part of the recipients field
+    should not themselves have the recipient field set.
+    """
+
+    def __init__(self, text, sound=None, recipients=None):
+        self.text = text
+        self.sound = sound
+        self.recipients = recipients
+
+    def __repr__(self):
+        return '<LayerClient.PushNotification "{text}">'.format(
+            text=self.text
+        )
+
+    def as_dict(self):
+        data = {
+            'text': self.text,
+        }
+        if self.sound:
+            data['sound'] = self.sound
+
+        # If per-recipient push notification instances are present, convert
+        # them to dictionaries as well. We don't simply recurse here to
+        # ensure that we do not have child PushNotifications with their own
+        # recipients fields.
+        if self.recipients:
+            recipients_dict = {}
+            for recipient, notification in self.recipients.iteritems():
+                recipients_dict[recipient] = {
+                    'text': notification.text,
+                    'sound': notification.sound,
+                }
+            data['recipients'] = self.recipients
+
+        return data
+
+
+class Conversation(BaseLayerResponse):
+    """
+    Represents a Layer conversation. Returned by the get_ and create_
+    conversation methods.
+    """
+
+    def __init__(self, id, url, messages_url=None, created_at=None,
+                 participants=[], distinct=False, metadata=None):
```

Comment:
If you leave participants as an array, any time you modify participants, it'll get persisted.  Probably better to just do 
`self.partcipants = participants or []` below so you don't have to worry about that